src/lib.rs

```diff
@@ -138,33 +138,44 @@ macro_rules! imap_deconstruct {
 mod test {
     use super::*;
 
-    fn host_library<T: ImplicitClone + Copy>(value: &T) -> T {
+    fn assert_implicit_clone<T: ImplicitClone>(value: &T) -> T {
+        value.clone()
+    }
+
+    fn assert_copy<T: Copy>(value: &T) -> T {
```

Comment:
now we are testing basic Rust functionality? lol
make it ImplicitClone + Copy i guess

Reply:
good point xD